redis-store/src/lib.rs

```diff
@@ -34,101 +24,12 @@ impl From<RedisStoreError> for session_store::Error {
     }
 }
 
-/// A Redis session store.
-#[derive(Debug, Clone, Default)]
-pub struct RedisStore<C: KeysInterface + Send + Sync> {
-    client: C,
-}
-
-impl<C: KeysInterface + Send + Sync> RedisStore<C> {
-    /// Create a new Redis store with the provided client.
-    ///
-    /// # Examples
-    ///
-    /// ```rust,no_run
-    /// use tower_sessions_redis_store::{fred::prelude::*, RedisStore};
-    ///
-    /// # tokio_test::block_on(async {
-    /// let pool = Pool::new(Config::default(), None, None, None, 6).unwrap();
-    ///
-    /// let _ = pool.connect();
-    /// pool.wait_for_connect().await.unwrap();
-    ///
-    /// let session_store = RedisStore::new(pool);
-    /// })
-    /// ```
-    pub fn new(client: C) -> Self {
-        Self { client }
-    }
+#[cfg(feature = "dynamic-pool")]
+pub mod dynamic_pool;
+#[cfg(feature = "dynamic-pool")]
```

Comment:
I think this will break feature unification since there are now two `RedisStore` types right?

Reply:
No, it just exports both the module `dynamic_pool` and for the convenience, its public items.

We can use it both:
```rust
use tower_sessions_redis_store::RedisStore;
// and
use tower_sessions_redis_store::dynamic_pool::RedisStore;
```

Also by enabling `dynamic-pool` feature, the generic implementation won't be loaded (due to the fact that both are feature gated and mutually exclusive).

Reply:
> mutually exclusive

Unfortunately this isn't really supported by Rust and is [strongly discouraged](https://doc.rust-lang.org/cargo/reference/features.html#mutually-exclusive-features).

Reply:
Your linked reference is not related to your concern and this PR. I meant the imports and codes are mutually exclusive - if you enable `dynamic-pool` features, the compiler will use only dynamic pool related imports/code from the `dynamic_pool` module and if you don't, it will import the generic.

Please check out the example, tests and CI for a better understanding. It's impossible to have both (generics and dynamic_pool module) imports with any features combination...